frontend/app/src/components/history/trades/ExternalTradeForm.vue

```diff
@@ -407,62 +388,27 @@ onMounted(() => {
                 @blur="v$.amount.$touch()"
               />
-              <div
-                :class="`external-trade-form__grouped-amount-input d-flex ${
-                  selectedCalculationInput === 'quoteAmount'
-                    ? 'flex-column-reverse'
-                    : 'flex-column'
-                }`"
-              >
-                <amount-input
-                  ref="rateInput"
-                  v-model="rate"
-                  :disabled="selectedCalculationInput !== 'rate'"
-                  :label="t('external_trade_form.rate.label')"
-                  :loading="fetching"
-                  :error-messages="v$.rate.$errors.map(e => e.$message)"
-                  data-cy="rate"
-                  :hide-details="selectedCalculationInput !== 'rate'"
-                  :class="`${
-                    selectedCalculationInput === 'rate'
-                      ? 'v-input--is-enabled'
-                      : ''
-                  }`"
-                  filled
-                  persistent-hint
-                  @blur="v$.rate.$touch()"
-                />
-                <amount-input
-                  ref="quoteAmountInput"
-                  v-model="quoteAmount"
-                  :disabled="selectedCalculationInput !== 'quoteAmount'"
-                  :error-messages="toMessages(v$.quoteAmount)"
-                  data-cy="quote-amount"
-                  :hide-details="selectedCalculationInput !== 'quoteAmount'"
-                  :class="`${
-                    selectedCalculationInput === 'quoteAmount'
-                      ? 'v-input--is-enabled'
-                      : ''
-                  }`"
-                  :label="t('external_trade_form.quote_amount.label')"
-                  filled
-                  @blur="v$.quoteAmount.$touch()"
-                />
-                <v-btn
-                  class="external-trade-form__grouped-amount-input__swap-button"
-                  fab
-                  small
-                  dark
-                  color="primary"
-                  data-cy="grouped-amount-input__swap-button"
-                  @click="swapAmountInput"
-                >
-                  <v-icon>mdi-swap-vertical</v-icon>
-                </v-btn>
-              </div>
+              <two-fields-amount-input
+                class="mb-5"
+                :primary-value.sync="rate"
+                :secondary-value.sync="quoteAmount"
+                :loading="fetching"
+                :disabled="fetching"
+                :primary-attrs="{
+                  'data-cy': 'rate',
+                  label: t('external_trade_form.rate.label'),
+                  errorMessages: v$.rate.$errors.map(e => e.$message)
+                }"
+                :secondary-attrs="{
+                  'data-cy': 'quote-amount',
+                  label: t('external_trade_form.quote_amount.label'),
+                  errorMessages: v$.quoteAmount.$errors.map(e => e.$message)
+                }"
```

Comment:
> Also maybe instead of primary and secondary attrs do a label= { primary?: string, secondary?: string }
and the same for errorMessages

Hmm for this we will need to list alll props manually, instead just using `useAttrs` inside.

and also for the data-cy, how we structure the prop then, if we want to put it for both this component, and the fields inside it.

Reply:
for data-cy you could add a `data-cy="trade-rate"` to the component in the form and just use a `data-cy="primary"` and a `data-cy="secondary"` for the internal fields

you should be able to use a `[data-cy="trade-rate"][data-cy="primary"]` selector in the e2e test to get it

Reply:
> > Also maybe instead of primary and secondary attrs do a label= { primary?: string, secondary?: string }
> > and the same for errorMessages
> 
> Hmm for this we will need to list all of the props manually, instead of just using `useAttrs` inside. Also for the `data-cy`, how do we structure the prop then, if we want to put it for both this component and the fields inside it?

So from what I have seen you pass the rootAttrs to both children, we don't need to remove that, we can just keep it as it is.

However from what I have seen primary-attrs and secondary-attrs are only used for two specific fields (other than data-cy) errorMessages and labels, correct?

Reply:
We can also alternately do `primary-label`, `secondary-label` and the same for error messages. Either approach is fine because it makes it clear that this is something the comp supports.

The only reason we use attrs for the rest of the vuetify components is that there is no easy way at least at the moment to make sure that the components picks the property types of the wrapped vuetify component.